lxd/device/nic_bridged.go

```diff
@@ -390,6 +390,13 @@ func (d *nicBridged) checkAddressConflict() error {
 			return nil
 		}
 
+		// Skip our own NIC if it is in the pending expanded config.
+		// This allows renaming a NIC without triggering a duplicate device error.
+		_, nicInPendingExpandedConfig := d.inst.ExpandedConfig()[nicName]
```

Comment:
Would it be worth it to move the `d.inst.ExpandedConfig` call inside the `if sameLogicalInstance`?